src/explore-education-statistics-frontend/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -24,156 +24,167 @@ interface Props {
+      }
     });
-  };
-
-  useEffect(() => {
-    const sortedFilters = sortBy(
-      [...removeSiblinglessTotalRows(filters), locations],
-      [options => options.length],
-    );
 
-    const halfwayIndex = Math.floor(sortedFilters.length / 2);
+    const removeSiblinglessTotalRows = (
+      categoryFilters: Dictionary<CategoryFilter[]>,
+    ): CategoryFilter[][] => {
+      return Object.values(categoryFilters).filter(filter => {
+        return filter.length > 1 || !filter[0].isTotal;
+      });
+    };
 
-    setTableHeaders({
-      columnGroups: sortedFilters.slice(0, halfwayIndex),
-      rowGroups: sortedFilters.slice(halfwayIndex),
-      columns: timePeriods,
-      rows: indicators,
-    });
-  }, [filters, timePeriods, locations, indicators]);
-
-  const columnHeaders: string[][] = [
-    ...tableHeaders.columnGroups.map(colGroup =>
-      colGroup.map(group => group.label),
-    ),
-    tableHeaders.columns.map(column => column.label),
-  ];
-
-  const rowHeaders: string[][] = [
-    ...tableHeaders.rowGroups.map(rowGroup =>
-      rowGroup.map(group => group.label),
-    ),
-    tableHeaders.rows.map(row => row.label),
-  ];
-
-  const rowHeadersCartesian = cartesian(
-    ...tableHeaders.rowGroups,
-    tableHeaders.rows,
-  );
-
-  const columnHeadersCartesian = cartesian(
-    ...tableHeaders.columnGroups,
-    tableHeaders.columns,
-  );
-
-  const rows = rowHeadersCartesian.map(rowFilterCombination => {
-    const rowCol1 = last(rowFilterCombination);
-
-    return columnHeadersCartesian.map(columnFilterCombination => {
-      const rowCol2 = last(columnFilterCombination);
-
-      // User could choose to flip rows and columns
-      const indicator = (rowCol1 instanceof Indicator
-        ? rowCol1
-        : rowCol2) as Indicator;
-      const timePeriod = (rowCol2 instanceof TimePeriod
-        ? rowCol2
-        : rowCol1) as TimePeriod;
-
-      const filterCombination = [
-        ...rowFilterCombination,
-        ...columnFilterCombination,
-      ];
-
-      const categoryFilters = filterCombination.filter(
-        filter => filter instanceof CategoryFilter,
+    useEffect(() => {
+      const sortedFilters = sortBy(
+        [...removeSiblinglessTotalRows(filters), locations],
+        [options => options.length],
       );
 
-      const locationFilters = filterCombination.filter(
-        filter => filter instanceof LocationFilter,
-      ) as LocationFilter[];
-
-      const matchingResult = results.find(result => {
-        return (
-          categoryFilters.every(filter =>
-            result.filters.includes(filter.value),
-          ) &&
-          result.timePeriod === timePeriod.value &&
-          locationFilters.every(
-            filter =>
-              result.location[filter.level] &&
-              result.location[filter.level].code === filter.value,
-          )
-        );
+      const halfwayIndex = Math.floor(sortedFilters.length / 2);
+
+      setTableHeaders({
```

Comment:
Shouldn't need to be re-sorting any of the table headers as we've already done this in `TableToolPage`. This entire `useEffect` is redundant now